internal/js-ast-utils/isFunctionNode.test.ts

```diff
@@ -0,0 +1,60 @@
+import {test} from "rome";
+import {isFunctionNode} from "./isFunctionNode";
+import {template} from "./template";
+import {createBuilder} from "@internal/ast/utils";
+import {
+	AnyNode,
+	jsClassDeclaration,
+	jsVariableDeclarationStatement,
+} from "@internal/ast";
+
+test(
+	"returns true for all function node types",
+	(t) => {
+		t.true(isFunctionNode(template.statement`function foo(){}`));
+		t.true(
+			isFunctionNode(
+				(jsVariableDeclarationStatement.assert(
+					template.statement`const foo = function () { }`,
+				).declaration.declarations[0].init as AnyNode),
```

Comment:
We shouldn't be using `as` casts unless under extremely restricted circumstances where we would have otherwise used a `// @ts-ignore`. They hide some pretty gnarly errors and reduce the type safety we have with our fully typed AST. I assume you need to do it here because `init` can be `undefined`. There isn't any value to having everything wrapped in a variable statement anyway so you can just make this an expression:

```suggestion
				template.expression`(function () {})`,
```

Reply:
yes i had to cast this one becuase init can be undefined and yeah i agree that variable statement doesn't have any value here so will change it simple expression